### PR TITLE
Use package_data for package binaries instead of scripts

### LINKED ISSUE #2054: semgrep thows exception on python setup.py install
**Describe the bug**
```
Processing semgrep-0.31.1-cp36.cp37.cp38.py36.py37.py38-none-macosx_10_14_x86_64.whl
Installing semgrep-0.31.1-cp36.cp37.cp38.py36.py37.py38-none-macosx_10_14_x86_64.whl to /Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages
Adding semgrep 0.31.1 to easy-install.pth file
Traceback (most recent call last):
  File "setup.py", line 56, in <module>
    install_requires=requirements,
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/__init__.py", line 144, in setup
    return distutils.core.setup(**attrs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/install.py", line 117, in do_egg_install
    cmd.run(show_deprecation=False)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 424, in run
    self.easy_install(spec, not self.no_deps)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 673, in easy_install
    return self.install_item(None, spec, tmpdir, deps, True)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 720, in install_item
    self.process_distribution(spec, dist, deps)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 765, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/pkg_resources/__init__.py", line 783, in resolve
    replace_conflicting=replace_conflicting
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1066, in best_match
    return self.obtain(req, installer)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1078, in obtain
    return installer(requirement)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 692, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 720, in install_item
    self.process_distribution(spec, dist, deps)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 745, in process_distribution
    self.install_egg_scripts(dist)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 619, in install_egg_scripts
    dist.get_metadata('scripts/' + script_name)
  File "/Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1426, in get_metadata
    return value.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 0: invalid continuation byte in scripts/spacegrep file at path: /Users/ajinabraham/Code/Mobile-Security-Framework-MobSF/v/lib/python3.7/site-packages/semgrep-0.31.1-py3.7-macosx-10.9-x86_64.egg/EGG-INFO/scripts/spacegrep
````

**To Reproduce**
Steps to reproduce the behavior, ideally a link to https://semgrep.dev:

Add semgrep to a setup.py script and run `python setup.py install`

**Expected behavior**
install gracefully

**Screenshots**

**What is the priority of the bug to you?**
NA

**Environment**
pypi, 0.31.1

### PR DESCRIPTION
**Pull request by mschwager**
Fixes https://github.com/returntocorp/semgrep/issues/2054.

There will undoubtedly be build failures I'll have to fix, I'm just putting this up to get early eyes on if I'm missing anything obvious, or obvious reasons why this won't work. I won't get into _all_ the details as to why I chose `package_data` for the binaries instead of `scripts` - comments in #2054 cover that.

**TL;DR is [`package_data`](https://packaging.python.org/guides/distributing-packages-using-setuptools/#package-data) is a much cleaner solution for including arbitrary data files (including binaries), and allows us to easily reference the resultant installed file via `pkg_resources`.** We will no longer have to muck around with system binaries/scripts, as [`scripts`](https://packaging.python.org/guides/distributing-packages-using-setuptools/#scripts)/`install_scripts` does, and we can instead directly reference files inside the installed package.

Potential buggy areas:

* There is now more opportunity for a `semgrep` : `semgrep-core` version mismatch. E.g. `semgrep` is somehow missing the `package_data`, falls back on an old `semgrep-core` version, and the version mismatch causes runtime issues. I believe this issue is highly unlikely to occur in practice, and existed before but wasn't explicitly defined. I think a better solution in the long-term is to avoid looking for system-level binaries and only reference `package_data` unless an env variable pointing to `semgrep-core` is explicitly defined. I believe we can live with this deficiency for now, and move toward a cleaner, easier to extend/improve/understand solution right now.

I tested this out with a local `.whl` file install, and a mocked PyPI install via https://github.com/steiza/simplepypi - both worked as expected. I also expect this to solve @blshkv's problem in https://github.com/returntocorp/semgrep/issues/2054.

### CODE REVIEW

**Comment by mschwager** [comment]
Note this will also require brew formula changes.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -95,6 +95,8 @@ jobs:
       - name: Install Tox and any other packages
         run: pip install pipenv==2018.11.26 wheel==0.34.2 tox==3.15.0
       - name: Run Tox
+        env:
+          SEMGREP_SKIP_BIN: true
```
Comment (mschwager) [inline .github/workflows/tests.yml:99]: This replaces the `if "TOX_ENV_NAME" in os.environ` check before.

**Comment by brendongo** [comment]
Doesn't look like you're missing anything. FYI HOMEBREW_SYSTEM is set automatically when doing a `brew install` but no reason we can't just set the new environment var in the brew formula.

**Review by brendongo — APPROVED** [review]

**Comment by mschwager** [comment]
> FYI HOMEBREW_SYSTEM is set automatically when doing a `brew install` but no reason we can't just set the new environment var in the brew formula.

That's what I figured. I thought it'd be better to have a single, explicit env variable we provide rather than try to enumerate all the environments we _don't_ want to build the binaries in our setup script.
